### PR TITLE
Domain event bus refactor

### PR DESCRIPTION
**Pull request by rafamarquezv**
Domain event bus refactor to be able to dispatch side effects from observers.

### CODE REVIEW

File: src/openads/domain/service/DomainEventBus.js [hunk 1]
```diff
@@ -12,11 +12,28 @@ export default class DomainEventBus {
...
   }
 
-  static raise ({domainEvent}) {
-    this._observers.get(domainEvent.eventName).forEach(observer => observer({payload: domainEvent.payload}))
+  raise ({domainEvent}) {
+    this._observers
+      .get(domainEvent.eventName)
+      .forEach(observer => observer({
+        payload: domainEvent.payload,
+        dispatcher: (domainEvent.dispatcher) ? () => this.raise({domainEvent: domainEvent.dispatcher}) : undefined
```
[the remark this answers is not in the document]
Comment (rafamarquezv) [reply src/openads/domain/service/DomainEventBus.js]: Refactored. ^_^

File: src/openads/domain/service/DomainEventBus.js [hunk 2]
```diff
@@ -12,11 +12,28 @@ export default class DomainEventBus {
...
     }
-    this._observers.get(eventName).push(observer)
   }
 
-  static raise ({domainEvent}) {
-    this._observers.get(domainEvent.eventName).forEach(observer => observer({payload: domainEvent.payload}))
+  raise ({domainEvent}) {
+    this._observers
+      .get(domainEvent.eventName)
+      .forEach(observer => observer({
```
Comment (alextremp) [inline src/openads/domain/service/DomainEventBus.js]: Just a question:
The observers will execute in order sequentially and if one fails we will face that all the observers chain will fail. Is that the expected behavior?

Comment (victuxbb) [reply src/openads/domain/service/DomainEventBus.js]: You are right. We need a try catch block wrapping the observer to not stop execution. 
Also in the catch block will be great have some kind of log...

Comment (rafamarquezv) [reply src/openads/domain/service/DomainEventBus.js]: Done

**Pull request closed without merging**